fieldmask/update.go

```diff
@@ -46,6 +46,16 @@ func Update(mask *fieldmaskpb.FieldMask, dst, src proto.Message) {
 }
 
 func updateWireSetFields(dst, src protoreflect.Message) {
+	// handle oneof fields that have changed between src and dst
+	oneofs := dst.Descriptor().Oneofs()
+	for i := 0; i < oneofs.Len(); i++ {
+		oneof := oneofs.Get(i)
+		whichDst := dst.WhichOneof(oneof)
+		whichSrc := src.WhichOneof(oneof)
+		if whichDst != whichSrc && whichSrc != nil {
+			dst.Set(whichSrc, src.Get(whichSrc))
+		}
+	}
 	src.Range(func(field protoreflect.FieldDescriptor, value protoreflect.Value) bool {
```

Comment:
do we need to account for oneof fields down here too? i.e. ship oneof fields that were handled in the code above?

Reply:
It's is not needed. It's essentially a noop in the `src.Range` block as the fields have already been updated, and I'd argue that the overhead of keeping track of it is not worth it.

Reply:
As we discussed, I'm planning to add a mother-of-all proto messages in this repository so we can get proper test coverage for more edgy cases like this.

Reply:
Makes sense, I get it.